### ISSUE TITLE
🐛 overrides[].includes not resolved from project root for nested packages using extends: "//"

### ISSUE THREAD

**Issue by dylanmoz** [biomejs/biome#11122]
### Environment information

Biome version: 2.5.6 (also reproduces on 2.5.1)

### What happened?

Reproduction: https://github.com/dylanmoz/biome-nested-overrides-includes-repro

### Summary

In a monorepo, glob patterns in a **root config's `overrides[].includes`** are matched against the **nested package's directory**, not the project root, when the package has a nested config with `"root": false, "extends": "//"`.

As a result, a root-level override written with a project-root-relative path (e.g. `./packages/*/src/**`) silently fails to match any file inside a nested package. The same override works correctly for packages that do **not** have a nested config.

### Reproduction

```bash
git clone https://github.com/dylanmoz/biome-nested-overrides-includes-repro
cd biome-nested-overrides-includes-repro
npm install
npm run s1   # the bug
npm run s2   # confirms the resolution base
```

**Scenario 1 (`scenario-1-overrides-nested/`)** — root `biome.json` enables `useFilenamingConvention` (kebab-case) and disables it via one override:

```jsonc
"overrides": [
  { "includes": ["./packages/*/src/**"],
    "linter": { "rules": { "style": { "useFilenamingConvention": "off" } } } }
]
```

Two packages each contain a badly-named `src/MyBadName.ts`:

- `packages/pkg-a-nested/` — has a nested `biome.json` (`{ "root": false, "extends": "//" }`)
- `packages/pkg-b-plain/` — has **no** nested config

**Scenario 2 (`scenario-2-workaround/`)** — the same nested package, but the root override is written relative to the package directory (`"includes": ["src/**"]`).

### Expected behavior

In scenario 1, the override matches both files (both live at `packages/<name>/src/`), so neither is reported.

### Actual behavior

`npm run s1`:

```
packages/pkg-a-nested/src/MyBadName.ts lint/style/useFilenamingConvention
  × The filename should be in kebab-case.
Found 1 error.
```

Only `pkg-a-nested` is reported; `pkg-b-plain` is correctly suppressed. The only difference between the two packages is the nested `extends: "//"` config, which isolates the cause to nested-config resolution.

`npm run s2` reports no errors — writing the override relative to the package dir (`src/**`) makes it apply. This confirms the override glob is resolved relative to the nested package's directory rather than the project root.

### Expected result

The override in the root biome.json matches both files (both live at `packages/<name>/src/`), so neither file is reported as an lint error.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by suryarupi** [comment]
Hi! I'd like to investigate this issue and work on a fix if it's still available. I can reproduce the behavior from the reproduction repository, and I'll start by tracing how overrides[].includes paths are resolved when a nested configuration uses "root": false and "extends": "//".
